txlib/api/resources.py

```diff
@@ -24,19 +25,59 @@ class Resource(BaseModel):
     }
     url_fields = {'project_slug', 'slug'}
 
+    BINARY_TYPES = ['DOCX', 'XLSX']
+
+    def _is_binary_file(self):
+        if hasattr(self, 'i18n_type'):
+            return self.i18n_type in self.BINARY_TYPES
+        return False
+
     def retrieve_content(self):
         """Retrieve the content of a resource."""
         path = self._construct_path_to_source_content()
         res = self._http.get(path)
         self._populated_fields['content'] = res['content']
         return res['content']
 
+    def _act_on_temp_file(self, action, path, file_content, kwargs):
```

Comment:
It is not clear why to act on a temporary file, can you extend the description? Also, please add description for the arguments.

Reply:
You don't need a temporary file. I assume `_http` is a wrapper around `requests`. If that's the case, take a look at this

```python
import requests
from io import BytesIO

response = requests.post(
    'https://www.transifex.com/api/2/project/kb1/resources',
    auth=('api', "1/c069b0031b8da0be45309f7f8e0ad7994e48f41a"),
    data={'slug': "from_api", 'name': "from_api", 'i18n_type': "KEYVALUEJSON"},
    files={'foo': BytesIO(b'{"hello": "world"}')},
)

print(response.status_code)
# 201

print(response.text)
# '[1, 0, 0]'
```

Reply:
Nice touch @kbairak! In the original implementation I also spotted the clumsy solution with the temp file but due to my limited Python knowledge and having delayed I didn't pursue it more. Now I know that I should have asked for better ideas :)